homeassistant/components/updater/binary_sensor.py

```diff
@@ -32,31 +34,46 @@ def unique_id(self) -> str:
     @property
     def is_on(self) -> bool:
         """Return true if the binary sensor is on."""
-        return "on" if self._updater.update_available else "off"
-
-    @property
-    def device_class(self) -> str:
-        """Return the class of this device, from component DEVICE_CLASSES."""
-        return "connectivity"
+        return self._update_available
 
     @property
     def available(self) -> bool:
         """Return True if entity is available."""
-        return True if self._updater.update_available is not None else False
+        return self._update_available is not None
 
     @property
     def should_poll(self) -> bool:
         """Return True if entity has to be polled for state."""
-        return True
+        return False
 
     @property
     def device_state_attributes(self) -> dict:
         """Return the optional state attributes."""
         data = super().device_state_attributes
         if data is None:
             data = {}
-        if self._updater.release_notes:
-            data[ATTR_RELEASE_NOTES] = self._updater.release_notes
-        if self._updater.newest_version:
-            data[ATTR_NEWEST_VERSION] = self._updater.newest_version
+        if self._release_notes:
+            data[ATTR_RELEASE_NOTES] = self._release_notes
+        if self._newest_version:
+            data[ATTR_NEWEST_VERSION] = self._newest_version
         return data
+
+    async def async_added_to_hass(self):
+        """Register update dispatcher."""
+
+        async def async_state_update(updater: Updater):
```

Comment:
Make this a callback instead. Replace `async def` with `def` and decorate it with `@callback` imported from core.py.

Reply:
Done. What's the difference?

Reply:
nvm, got it

Reply:
The change applied to the wrong method. `async_added_to_hass` should still be a coroutine.

Callbacks have less overhead than coroutines.